near-sdk/src/types/event.rs

```diff
@@ -0,0 +1,84 @@
+pub trait EventJson {
+    type EventString: std::convert::AsRef<str>;
```

Comment:
My previous comment was referring to changing the name of the associated type, not the trait name, but I suppose this is fine. Only thing I'll mention is that it is ambiguous what the `EventString` is referring to -- whether it be the whole event message or just the JSON data